harness/determined/integrations/__init__.py

```diff
@@ -0,0 +1 @@
+from determined.integrations.huggingface.hf_callback import DetCallback
```

Comment:
The code layout here needs some updating.

We have a [code-layout RFC](https://github.com/rb-determined-ai/pedl/blob/9bd2168ec23fbba866194f3d7d3cfc243d267100/python_code_layout.md), which explains why our (newer) code is laid out the way it is.

There's a few reasons why this is messy:
- There should be one clear way that end users access a symbol.  But the naming of modules does not make it obvious to end users which of the following are most correct:
```python
from determined.integrations import DetCallback
from determined.integrations.huggingface import DetCallback
from determined.integrations.huggingface.hf_callback import DetCallback
```
- Importing huggingface code automatically as part of `determined.integrations` means that if there was another non-transformers integration, you wouldn't be able to import it unless you had `transformers` installed.
- Why is it called `determined.integrations.huggingface` instead of `determined.integrations.transformers`?
- Why put the huggingface integration under `det.integrations` instead of `det.transformers` or `det.pytorch.transformers`?  (I like flat hierarchies, so I think I'm most in favor of `det.transformers`)

Reply:
Agree w/ this all, and would love a very short import path.

Reply:
Added. It is now `determined.transformers.DetCallback` with a _ file for the name so there is only one way to import it afaik.

Reply:
looks good, thank you